lib/dom.js

```diff
@@ -1919,12 +2067,26 @@ function _insertBefore(parent, node, child, _inDocumentAssertion) {
 	}
 	do {
 		newFirst.parentNode = parent;
+		if (parent.nodeType === parent.ELEMENT_NODE) {
+			newFirst.parentElement = parent;
+		} else {
+			newFirst.parentElement = null;
+		}
 	} while (newFirst !== newLast && (newFirst = newFirst.nextSibling));
 	_onUpdateChild(parent.ownerDocument || parent, parent);
-	//console.log(parent.lastChild.nextSibling == null)
 	if (node.nodeType == DOCUMENT_FRAGMENT_NODE) {
 		node.firstChild = node.lastChild = null;
 	}
+
+	// find out if node is connected to a document
+	var parentNode = parent;
+	do {
+		if (parentNode.nodeType === parentNode.DOCUMENT_NODE) {
+			node.isConnected = true;
+			break;
+		}
+	} while ((parentNode = parentNode.parentNode));
+
```

Comment:
Same here regarding implementing it as a getter, since this can be quite some iterations in big documents even during parsing, and the property would also need to be maintained as part of other operations for every node in the tree, even when it is never requested.

And I think the implementation should just use the new `getRootNode` implementation and check it's `nodeType`